go/engine/deprovision_test.go

```diff
@@ -342,3 +415,32 @@ func TestCurrentDeviceRevoked(t *testing.T) {
 	tc.G.SecretStoreAll = nil
 	assertCurrentDeviceRevoked(tc)
 }
+
+func TestDeprovisionLastDevice(t *testing.T) {
+	testDeprovisionLastDevice(t, false)
+}
+
+func TestDeprovisionLastDevicePUK(t *testing.T) {
+	testDeprovisionLastDevice(t, true)
+}
+
+// A user should be able to revoke all of their devices.
+func testDeprovisionLastDevice(t *testing.T, upgradePerUserKey bool) {
+	tc := SetupEngineTest(t, "deprovision")
+	defer tc.Cleanup()
+	tc.Tp.UpgradePerUserKey = upgradePerUserKey
+	if tc.G.SecretStoreAll == nil {
+		t.Fatal("Need a secret store for this test")
+	}
+	fu := assertDeprovisionWithSetup(tc, assertDeprovisionWithSetupArg{
+		revokePaperKey: true,
+	})
+	assertNumDevicesAndKeys(tc, fu, 0, 0)
+
+	// Now, test deprovision codepath with no secret store
+	tc.G.SecretStoreAll = nil
+	fu = assertDeprovisionWithSetup(tc, assertDeprovisionWithSetupArg{
+		revokePaperKey: true,
+	})
+	assertNumDevicesAndKeys(tc, fu, 0, 0)
+}
```

Comment:
Is there a test that tests making a device, revoking a device, and then adding a new device that has to follow the prevs, and for that device to use the prev decryption logic to follow back to the previous keys?